### PR TITLE
Add more fields to fluentbit spec

### PR DESCRIPTION
**Pull request by euclidgame**

### CODE REVIEW

File: src/controller_examples/fluent_controller/fluentbit/exec/resource/daemon_set.rs [hunk 1]
```diff
@@ -226,7 +236,7 @@ fn make_fluentbit_pod_spec(fb: &FluentBit) -> (pod_spec: PodSpec)
                 });
                 proof {
                     assert_seqs_equal!(
-                        volume_mounts@.map_values(|volume_mount: VolumeMount| volume_mount@),
+                        volume_mounts@.map_values(|volume_mount: VolumeMount| volume_mount@) ==
```
Comment (marshtompsxd) [inline src/controller_examples/fluent_controller/fluentbit/exec/resource/daemon_set.rs:239]: What is this change for?

File: src/controller_examples/fluent_controller/fluentbit/exec/resource/daemon_set.rs [hunk 2]
```diff
@@ -147,6 +147,7 @@ pub fn make_daemon_set(fb: &FluentBit) -> (daemon_set: DaemonSet)
     daemon_set
 }
 
+#[verifier(spinoff_prover)]
```
Comment (marshtompsxd) [inline src/controller_examples/fluent_controller/fluentbit/exec/resource/daemon_set.rs:150]: Would it be slower without spinoff?

Comment (euclidgame) [reply src/controller_examples/fluent_controller/fluentbit/exec/resource/daemon_set.rs:150]: Yes on my laptop

File: src/controller_examples/fluent_controller/fluentbit/exec/resource/daemon_set.rs (hunk 1 above)
Comment (euclidgame) [reply src/controller_examples/fluent_controller/fluentbit/exec/resource/daemon_set.rs:239]: Sorry. I changed `assert_seqs_equal!` to `assume` before, so I also made this change.

Comment (marshtompsxd) [reply src/controller_examples/fluent_controller/fluentbit/exec/resource/daemon_set.rs:239]: And I just realized that assert_seqs_equal supports both styles. Could you revert this back for now?

File: src/deps_hack/src/lib.rs [hunk 3]
```diff
@@ -207,7 +207,15 @@ pub struct FluentBitSpec {
     pub labels: std::collections::BTreeMap<String, String>,
     #[serde(default)]
     pub annotations: std::collections::BTreeMap<String, String>,
+    #[serde(default, rename = "serviceAccountAnnotations")]
+    pub service_account_annotations: std::collections::BTreeMap<String, String>,
+    #[serde(default, rename = "serviceLabels")]
+    pub service_labels: Option<std::collections::BTreeMap<String, String>>,
+    #[serde(default, rename = "serviceAnnotations")]
+    pub service_annotations: Option<std::collections::BTreeMap<String, String>>,
```
Comment (marshtompsxd) [inline src/deps_hack/src/lib.rs:215]: I think service_annotations and service_labels don't need to be optional.

File: src/controller_examples/fluent_controller/fluentbit/exec/resource/daemon_set.rs [hunk 4]
```diff
@@ -186,38 +190,44 @@ fn make_fluentbit_pod_spec(fb: &FluentBit) -> (pod_spec: PodSpec)
...
-                let mut volume_mounts = Vec::new();
-                volume_mounts.push({
-                    let mut volume_mount = VolumeMount::default();
-                    volume_mount.set_name(new_strlit("varlibcontainers").to_string());
-                    volume_mount.set_read_only(true);
-                    volume_mount.set_mount_path(new_strlit("/containers").to_string());
-                    volume_mount.overwrite_mount_propagation(fb.spec().internal_mount_propagation());
-                    volume_mount
-                });
+                let mut volume_mounts = if fb.spec().volume_mounts().is_some() { fb.spec().volume_mounts().unwrap() } else { Vec::new() };
```
Comment (marshtompsxd) [inline src/controller_examples/fluent_controller/fluentbit/exec/resource/daemon_set.rs:193]: Could you format this line to
```
if fb.spec().volume_mounts().is_some() {
  fb.spec().volume_mounts().unwrap()
} else {
  Vec::new()
};
```

File: src/controller_examples/fluent_controller/fluentbit/exec/resource/service.rs [hunk 5]
```diff
@@ -119,8 +119,16 @@ pub fn make_service(fb: &FluentBit) -> (service: Service)
...
         metadata.set_name(make_service_name(fb));
         metadata.set_owner_references(make_owner_references(fb));
-        metadata.set_labels(make_labels(fb));
-        metadata.set_annotations(fb.spec().annotations());
+        metadata.set_labels({
+            if fb.spec().service_labels().is_some() {
+                fb.spec().service_labels().unwrap()
+            } else { make_labels(fb) }
+        });
+        if fb.spec().service_annotations().is_some() {
```
Comment (marshtompsxd) [inline src/controller_examples/fluent_controller/fluentbit/exec/resource/service.rs:127]: For the service annotations, I think we can just set it to `fb.spec().service_annotations()`, similar to what you did for service account.

File: src/deps_hack/src/lib.rs (hunk 3 above)
Comment (marshtompsxd) [reply src/deps_hack/src/lib.rs:215]: The reference controller is doing
```
	if len(fb.Spec.Service.Labels) != 0 {
		labels = fb.Spec.Service.Labels
	} else {
		labels = fb.Labels
	}
```
so I think you can add a `len()` method to our StringMap to implement this logic, without making service_labels optional

**Pull request merged**